code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -30,6 +30,8 @@
 	var/target_temperature = T20C
 	var/heat_capacity = 0
 	var/interactive = TRUE // So mapmakers can disable interaction.
+	var/allows_cooling = TRUE // Non anomalous thermomachines are always allowed to cool
+	var/has_anomaly_core = FALSE
```

Comment:
You should use a direct reference to the anomaly core instead.

Reply:
If you mean in regards to allows_cooling it is distinct because with the addition of the slotting in capability it retains cooling while if you make the anomalous version it disallows

Reply:
Updated has_anomaly_core to has_pyro_anomaly_core